src/main/java/com/pinterest/secor/uploader/FormatConvertingUploadManager.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.secor.uploader;
+
+import com.pinterest.secor.common.LogFilePath;
+import com.pinterest.secor.common.SecorConfig;
+import com.pinterest.secor.io.FileReader;
+import com.pinterest.secor.io.FileWriter;
+import com.pinterest.secor.io.KeyValue;
+import com.pinterest.secor.uploader.UploadManager;
+import com.pinterest.secor.util.CompressionUtil;
+import com.pinterest.secor.util.FileUtil;
+import com.pinterest.secor.util.IdUtil;
+import com.pinterest.secor.util.ReflectionUtil;
+
+import java.io.File;
+
+import org.apache.hadoop.io.compress.CompressionCodec;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages uploads using the configured upload manager
+ * Converts the local files to the specified format before upload
+ *
+ * @author Jason Butterfield (jason.butterfield@outreach.io)
+ */
+public class FormatConvertingUploadManager extends UploadManager {
+    private static final Logger LOG = LoggerFactory.getLogger(FormatConvertingUploadManager.class);
+
+    private UploadManager mUploadManager;
+
+    public FormatConvertingUploadManager(SecorConfig config) throws Exception {
+        super(config);
+        mUploadManager = createUploadManager();
+    }
+
+    @Override
+    public TempFileUploadHandle<Handle<?>> upload(LogFilePath localPath) throws Exception {
+        // convert the file from the internal format to the external format
+        System.out.println("localPath: " + localPath);
+        System.out.println("mUploadManager: " + mUploadManager);
+        LogFilePath convertedFilePath = convertFile(localPath);
+
+        // The TempFileUploadHandle will delete the temp file after it resolves
+        return new TempFileUploadHandle(mUploadManager.upload(convertedFilePath), convertedFilePath);
+    }
+
+    /**
+     * This method converts a file 
+     * Exposed for testing only
+     */
+    public LogFilePath convertFile(LogFilePath srcPath) throws Exception {
+        FileReader reader = null;
+        FileWriter writer = null;
+        int copiedMessages = 0;
+
+        String localConvertedPrefix = mConfig.getLocalPath() + '/' +
```

Comment:
Nit: Ruby's got a gem that lets you join paths together and handles any extra or missing "/"s, does Java? And would it be safer to use that here? 

This don't matter so don't sweat it if you disagree or don't wanna put the time into it. Your code here looks pretty good so I'm just looking for ways to help.

Reply:
That's a great idea.  I shamelessly copies this from another file, but fixed it to use `Paths.get()` instead now.